flexible_event_config.md

```diff
@@ -143,24 +143,26 @@ In addition to the parameters that were added in Phase 1, we will add one additi
     // Represents a series of time windows, starting at the source registration time.
     // Reports for this spec will be delivered an hour after the end of each window.
     // Time is encoded as seconds after source registration.
+    // end_times should consist of strictly increasing integers without repetitions. 
     //
     // Note: specs with identical trigger_data cannot have overlapping windows;
     // this ensures that triggers match at most one spec. If event_report_windows
     // is omitted, will use the "event_report_window" or "event_report_windows" field specified at the global level for the source (or the default windows if none are specified).  End time is exclusive.
     "event_report_windows": {
-      "end_times": [<int>, ...]
+      "end_times": [<int>, ...],
     }
 
     // Represents an operator that summarizes the triggers within a window
     // count: number of triggers attributed within a window
     // value_sum: sum of the value of triggers within a window
     // The summary is reported as an index into a bucketization scheme.
     // Defaults to "count"
-    "summary_window_operator": <one of "count" or "value_sum">
+    "summary_window_operator": <one of "count" or "value_sum">, 
 
     // Represents a bucketization of the integers from [0, MAX_INT], encoded as
     // a list of integers where new buckets begin (excluding 0 which is
     // implicitly included).
+    // It should consist of strictly increasing positive integers without repetitions. 
```

Comment:
Similar to the `end_times` field, what is the benefit of requiring this? It seems like more work for the implementation to enforce the strict-increasing part when the bucket endpoints are already effectively a set.

Reply:
Major motivation has been mention in previous comments. Implementation-wise, we need pre-processing anyway, either to process it to reasonable window or just return error.

Reply:
> Major motivation

I disagree that the lack of a business use-case for duplicates or unordered values is a major motivation. We don't currently prevent duplicates in other areas where they are irrelevant, e.g. the source's `destination` field or `filter_data`'s list elements.

Reply:
Although implemented as list, both destination and filter_data are technically hash-table style and each element performs independently. However, end_times and summary_buckets are list, and relationship between the element are the key to it function.